build/main_test.go

```diff
@@ -34,3 +34,45 @@ func ExampleTransactionBuilder() {
 	fmt.Printf("tx base64: %s", txeB64)
 	// Output: tx base64: AAAAADZY/nWY0gx6beMpf4S8Ur0qHsjA8fbFtBzBx1cbQzHwAAAAZAAAAAAAAAABAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAALSRpLtCLv2eboZlEiHDSGR6Hb+zZL92fbSdNpObeE0EAAAAAAAAAAB3NZQAAAAAAAAAAARtDMfAAAABA2oIeQxoJl53RMRWFeLB865zcky39f2gf2PmUubCuJYccEePRSrTC8QQrMOgGwD8a6oe8dgltvezdDsmmXBPyBw==
 }
+
+// ExamplePathPayment creates and signs a simple transaction with PathPayment operation, and then
+// encodes it into a base64 string capable of being submitted to stellar-core.
+func ExamplePathPayment() {
```

Comment:
Personally, I don't like it.  Conceptually, it replaces the existing `Amount` mutators (in a path payment context) which is one thing I was trying to avoid.  Think "Small pieces working together", something like the unix philosophy.

IMO, a payment within the stellar network is defined by three sets of information:  The destination information, the source information, and the intermediary path information.  I think that by having these three as separate mutators/concepts gives us the most flexibility and will let more feature-rich mutators be built through composition.

---

Also, one quick aside.  I've tried to replace "SendMax" in its various forms with  "SourceMax", IMO it consistent with the notion that every payment has the mentioned three sets of info.

I found myself numerous times getting confused and misusing "source" when it should be "send", so I feel it's a good change to make higher level APIs more understandable.  Note:  The documentation should still mention it sets the "SendMax" field when talking about how the underlying xdr operation is mutated.

---

If you believe that people will confuse "source" and "destination", I think we should work to disambiguate rather than abandon the concept of the gradual onramp.  I've got a more drastic change that I'll write up an post momentarily, but let me also offer some augmentations of my proposal above.    Instead of `Source()` how about `PayWith` or `PayUsing`?:

```go
Payment(
  Destination{"GBDT3K42LOPSHNAEHEJ6AVPADIJ4MAR64QEKKW2LQPBSKLYD22KUEH4P"},
  CreditAmount{ "EUR", "GCPZJ3MJQ3GUGJSBL6R3MLYZS6FKVHG67BPAINMXL3NWNXR5S6XG657P", "100" },
  PayWith(CreditAmount{ "USD", "GAWSI2JO2CF36Z43UGMUJCDQ2IMR5B3P5TMS7XM7NUTU3JHG3YJUDQXA", "50" }),
  Path().Native().Credit("BTC", "GAHJZHVKFLATAATJH46C7OK2ZOVRD47GZBGQ7P6OCVF6RJDCEG5JMQBQ"),
)
``` 

Regarding the path: The "gradual onramp" goal is one reasons that I am consciously keeping "path" mutators separate from "source".  It allows someone to learn that "A payment can be made from one asset to another" without having to understand that the system can hop through multiple assets as well.